storage/store.go

```diff
@@ -2873,19 +2873,27 @@ func (s *Store) updateReplicationGauges() error {
 
 	timestamp := s.ctx.Clock.Now()
 
+	// Make a copy of all the current replicas so we don't need to hold onto
+	// the store lock.
+	var replicas []*Replica
```

Comment:
why not use `newStoreRangeSet`? That'll mean I have one less place to refactor in #9190.